conans/paths/package_layouts/package_editable_layout.py

```diff
@@ -25,25 +27,28 @@ def conanfile(self):
         """
         return os.path.join(self.conan(), CONANFILE)
 
-    def editable_package_layout_file(self):
-        return os.path.join(self.conan(), CONAN_PACKAGE_LAYOUT_FILE)
+    def editable_cpp_info(self):
+        if self._layout_file:
+            if os.path.isfile(self._layout_file):
+                return EditableCppInfo.load(self._layout_file)
+            else:
+                raise ConanException("Layout file not found: %s" % self._layout_file)
 
     def export(self):
-        raise RuntimeError("Operation not allowed on a package installed as editable")
+        raise ConanException("Operation not allowed on a package installed as editable")
 
     def export_sources(self):
-        raise RuntimeError("Operation not allowed on a package installed as editable")
+        raise ConanException("Operation not allowed on a package installed as editable")
 
     def source(self):
-        raise RuntimeError("Operation not allowed on a package installed as editable")
+        raise ConanException("Operation not allowed on a package installed as editable")
 
     def load_metadata(self):
-        raise RuntimeError("Operation not allowed on a package installed as editable")
+        raise ConanException("Operation not allowed on a package installed as editable")
 
     def package(self, pref):
         assert isinstance(pref, PackageReference)
         assert pref.ref == self._ref
-        raise RuntimeError("Operation not allowed on a package installed as editable")
 
     def package_metadata(self):
         # FIXME: I know that downstream there is an except for IOError
```

Comment:
Please, review the exception that is being thrown from this function, we have to take a look at the places where the `package_metadata` is being called

Reply:
Added exception again

Reply:
The exception is there, but it is an `IOError` exception. I really dislike throwing from this function an `IOError` exception (when everything else is a `ConanException`) just because the context this function is called:

```python
    # Metadata
    def load_metadata(self):
        try:
            text = load(self.package_metadata())
            return PackageMetadata.loads(text)
        except IOError:
            return PackageMetadata()
```

I'm ok with it if we are ok with it. I was just asking for confirmation or maybe modify that function to catch a ConanException too and throw it from here.